olive/passes/onnx/dynamic_to_fixed_shape.py

```diff
@@ -0,0 +1,119 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from typing import Any, Callable, Dict, List
+
+from olive.common.pydantic_v1 import root_validator
+from olive.hardware import AcceleratorSpec
+from olive.model import ONNXModelHandler
+from olive.model.utils import resolve_onnx_path
+from olive.passes.olive_pass import Pass
+from olive.passes.onnx.common import get_external_data_config, model_proto_to_olive_model
+from olive.passes.pass_config import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class DynamicToFixedShape(Pass):
+    """Preprocess ONNX model for quantization targeting QNN Execution Provider."""
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        config = {
+            "dim_param": PassConfigParam(
+                type_=List[str],
+                default_value=None,
+                required=False,
+                description=("Symbolic parameter name. Provide dim_value if specified."),
+            ),
+            "dim_value": PassConfigParam(
+                type_=List[int],
+                default_value=None,
+                required=False,
+                description=("Value to replace dim_param with in the model. Must be > 0."),
+            ),
+            "input_name": PassConfigParam(
+                type_=List[str],
+                default_value=None,
+                required=False,
+                description=("Model input name to replace shape of. Provide input_shape if specified."),
+            ),
+            "input_shape": PassConfigParam(
+                type_=List[List[int]],
+                default_value=None,
+                required=False,
+                description=(
+                    "Shape to use for input_shape. Provide comma separated list for the shape. "
+                    "All values must be > 0. e.g. [1,3,256,256]"
+                ),
+            ),
+        }
+        config.update(get_external_data_config())
+        return config
+
+    @staticmethod
+    def _validators() -> Dict[str, Callable[..., Any]]:
+        return {
+            "validate_dim_param_and_value": root_validator(allow_reuse=True)(_validate_dim_param_and_value),
```

Comment:
for root_validator, could we use only one?

Reply:
Updated. Also rebase from main branch to resolve the docs building error.